### ISSUE TITLE
Fcst.predict does not accept X_df with dynamic exogenous variables

### ISSUE THREAD

**Issue by melissafeeney** [Nixtla/mlforecast#307]
I am training a simple time series model, and my dataset includes 1 dynamic exogenous variable (holiday) and no static exogenous variables. The training set contains 131 weeks of data, and the test set contains 24 weeks of data. Each week starts on a Monday.

The training dataset looks like this:
<img width="392" alt="Screen Shot 2024-01-26 at 3 18 32 PM" src="https://github.com/Nixtla/mlforecast/assets/31778500/5c46b559-23b1-44b7-aa0a-212d8d2db757">

The test dataset looks like this:
<img width="359" alt="Screen Shot 2024-01-26 at 3 20 16 PM" src="https://github.com/Nixtla/mlforecast/assets/31778500/cc34f3eb-2eb0-4856-88c9-1bdbc785666b">

I am running this first train a forecasting model on my training data, after which I want to test it on my test data. Even though my test data contains unique_id, ds, and holiday (the dynamic exogenous variable), I get an error:
<img width="1292" alt="Screen Shot 2024-01-26 at 3 23 05 PM" src="https://github.com/Nixtla/mlforecast/assets/31778500/58ddf405-9588-4769-a563-80bc2ebe4b9d">

If I try to use the suggestions in the error to create X_df using fcst.make_future_dataframe(h) but when I do that, the ds dates transform into Sundays (instead of Mondays as they are in my train and test datasets).
<img width="277" alt="Screen Shot 2024-01-26 at 3 24 03 PM" src="https://github.com/Nixtla/mlforecast/assets/31778500/fcb3c329-07d1-4d29-a568-911a46174f5b">

If I try to use the suggestions in the error to use fcst.get_missing_future(h, X_df), I get this- and also notice how the dates changed from Monday into Sundays...:
<img width="565" alt="Screen Shot 2024-01-26 at 3 26 01 PM" src="https://github.com/Nixtla/mlforecast/assets/31778500/efb66fce-4153-4b3f-be58-fed3a0bdd397">

### Versions / Dependencies

I am using Google colab and installed MLforecast from pip as well as from source as suggested from another issue @298:
`! pip install git+https://github.com/Nixtla/mlforecast` 

version: mlforecast-0.11.6

### Reproduction script

```
from sklearn.ensemble import RandomForestRegressor

train = data.loc[0:130]
test = data.loc[131:]

models = RandomForestRegressor(random_state=0, n_estimators = 100)

fcst = MLForecast(models=models, freq = 'W')

fcst.fit(train, fitted = True, static_features=[])
preds = fcst.predict(h = 24, X_df = test[['unique_id', 'ds', 'holiday']]) 
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by melissafeeney** [comment]
Interestingly, changing the freq paramter when fitting the fcst object to 'W-MON' may have solved the issue! This runs without issue:

```
from sklearn.ensemble import RandomForestRegressor

train = data.loc[0:130]
test = data.loc[131:]

models = RandomForestRegressor(random_state=0, n_estimators = 100)

fcst = MLForecast(models=models, freq = 'W-MON')

fcst.fit(train, fitted = True, static_features=[])
preds = fcst.predict(h = 24, X_df = test[['unique_id', 'ds', 'holiday']]) 
```

**Comment by jmoralez** [comment]
Hey @melissafeeney, thanks for using mlforecast. We use the `freq` argument to build the future dates, so it must match the frequency of your series ('W-MON' in this case is the correct one). We're glad the debugging methods helped you figure out the problem.
